### PR TITLE
cjson: fix compiling static library on MSVC

### PR DESCRIPTION
**Pull request by Barakat**
cJSON static library compilation is broken on MSVC:

```
>meson compile -C build
...
C compiler for the host machine: cl (msvc 19.38.33133 "Microsoft (R) C/C++ Optimizing Compiler Version 19.38.33133 for x64")
C linker for the host machine: link link 14.38.33133.0
Host machine cpu family: x86_64
Host machine cpu: x86_64

Executing subproject cjson

cjson| Project name: cjson
cjson| Project version: 1.7.16
cjson| C compiler for the host machine: cl (msvc 19.38.33133 "Microsoft (R) C/C++ Optimizing Compiler Version 19.38.33133 for x64")
cjson| C linker for the host machine: link link 14.38.33133.0
cjson| Compiler for C supports arguments -Wno-deprecated-declarations: NO
cjson| Library m found: NO
cjson| Did not find pkg-config by name 'pkg-config'
cjson| Found pkg-config: NO
cjson| Build targets in project: 25
cjson| NOTICE: Future-deprecated features used:
cjson| * 0.64.0: {'copy arg in configure_file'}
cjson| Subproject cjson finished.

Build targets in project: 26

cjson-test undefined

  Subprojects
    cjson: YES 1 warnings

Found ninja-1.11.1 at "C:\Program Files\Meson\ninja.EXE"
Cleaning... 0 files.

...
[69/73] Linking target cjson-static.exe
FAILED: cjson-static.exe cjson-static.pdb
"link"  /MACHINE:x64 /OUT:cjson-static.exe cjson-static.exe.p/main.c.obj "/release" "/nologo" "/DEBUG" "/PDB:cjson-static.pdb" "subprojects\cJSON-1.7.16\libcjson.a" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
main.c.obj : error LNK2019: unresolved external symbol __imp_cJSON_Version referenced in function main
cjson-static.exe : fatal error LNK1120: 1 unresolved externals
...
```

This is a patch to fix the issue.

### CODE REVIEW

File: subprojects/packagefiles/cjson/meson.build [hunk 1]
```diff
@@ -6,12 +6,17 @@ project('cjson', 'c',
 dllimport_args = []
 cc = meson.get_compiler('c')
 if cc.get_argument_syntax() == 'msvc'
-  add_project_arguments('-D_CRT_SECURE_NO_WARNINGS', cc.get_supported_arguments('-Wno-deprecated-declarations'), language: 'c')
```
Comment (neheb) [inline subprojects/packagefiles/cjson/meson.build:9]: Why get rid of -Wno-deprecated-declarations?

Comment (Barakat) [reply subprojects/packagefiles/cjson/meson.build:9]: This flag a gcc/clang flag and not being recognized by MSVC:

```
cl : Command line error D8021 : invalid numeric argument '/Wdeprecated-declarations'
```

Comment (neheb) [reply subprojects/packagefiles/cjson/meson.build:9]: hmm? cc.get_supported_arguments should fail for MSVC and succeed for clang-cl.

Comment (Barakat) [reply subprojects/packagefiles/cjson/meson.build:9]: My bad. I have reverted this line.

**Pull request merged**